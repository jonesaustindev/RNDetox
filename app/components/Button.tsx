import React from 'react';
import {Text, TextStyle, TouchableOpacity, ViewStyle} from 'react-native';

interface RNTestButtonProps {
  text: string;
  testID?: string;
  buttonStyle?: ViewStyle;
  textStyle?: TextStyle;
  onPress?: () => void;
}

export function RNTestButton({
  text,
  testID,
  onPress,
  buttonStyle,
  textStyle,
}: RNTestButtonProps) {
  return (
    <TouchableOpacity
      testID={testID}
      onPress={onPress}
      activeOpacity={0.7}
      style={[
        {
          backgroundColor: '#A267AC',
          borderRadius: 6,
          height: 52,
          justifyContent: 'center',
          alignItems: 'center',
        },
        buttonStyle ? {...buttonStyle} : null,
      ]}>
      <Text
        style={[
          {fontSize: 18, color: '#ffffff'},
          textStyle ? {...textStyle} : null,
        ]}>
        {text && text}
      </Text>
    </TouchableOpacity>
  );
}
