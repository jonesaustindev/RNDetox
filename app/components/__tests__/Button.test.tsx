import React from 'react';
import {render} from '@testing-library/react-native';
import {RNTestButton} from '../Button';

it('renders button', () => {
  const button = render(
    <RNTestButton text="Testing" onPress={jest.fn} testID="testButton" />,
  );
  expect(button).toMatchSnapshot();
});
