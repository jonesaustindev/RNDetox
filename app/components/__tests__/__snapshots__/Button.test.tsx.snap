// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A267AC",
      "borderRadius": 6,
      "height": 52,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="testButton"
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        },
        null,
      ]
    }
  >
    Testing
  </Text>
</View>
`;
